api/app_sph_lms/api/view/course_view.py

```diff
@@ -1,55 +1,32 @@
 from app_sph_lms.api.serializer.course_serializer import (
     CourseCategorySerializer, CourseSerializer)
 from app_sph_lms.models import Category, Course, CourseCategory
-from app_sph_lms.utils.enum import StatusEnum
-from django_filters.rest_framework import DjangoFilterBackend
+from django.core.paginator import Paginator
 from rest_framework import generics
+from rest_framework.pagination import PageNumberPagination
+from rest_framework.response import Response
+
+class LargeResultsSetPagination(PageNumberPagination):
+    page_size = 12
+    page_size_query_param = 'page_size'
+    max_page_size = 12
+
+    def get_paginated_response(self, data):
+        return Response({
+            'page_size': self.page_size,
+            'count': self.page.paginator.count,
+            'totalPages': self.page.paginator.num_pages,
+            'current_page_number': self.page.number,
+            'next': self.get_next_link(),
+            'previous': self.get_previous_link(),
+            'results': data,
+        })
 
 
 class CourseList(generics.ListCreateAPIView):
-    queryset = Course.objects.all()
+    queryset = Course.objects.prefetch_related('coursecategory_set__category')
```

Comment:
I think this can be reverted to ` Course.objects.all()` since the `CourseSerializer` was recently updated to get related categories. (See here: [line 39 /api/serializer/course_serializer.py](https://github.com/framgia/sph-lms/blob/master/api/app_sph_lms/api/serializer/course_serializer.py))

Please test if it returns the same result.